services/speech/src/speechd/environment.js

```diff
@@ -0,0 +1,16 @@
+const exec = require("child_process").exec;
+
+const getExecutablePath = async () =>
+  new Promise((resolve, reject) => {
+    exec(`which speech-dispatcher`, (error, stdout) => {
```

Comment:
I'd never expect this to throw an error, unless `which` wasn't found in the `PATH`. I think it's more likely to return with an empty `stdout` if the `speech-dispatcher` isn't found.

Reply:
... I am *dead wrong* about this. `which` will return an exit code of `1` if the binary isn't found in `$PATH`.